### ISSUE TITLE
"Archive" of completed rooms such that when a new message is received the room is taken out of the archive

### ISSUE THREAD

**Issue by gabrc52** [matrix-org/matrix-spec#1705]
**Suggestion**

In many email clients there is an "Archive" feature, where users can either press a button or slide the email thread a certain way to signal that they are done with it and don't need to see it for the time being. Users can manually navigate to the archive to see these messages.

It would be nice to have this in Matrix. Of course, this would be best not as a client-side feature but stored in account data or somewhere various clients can all access and agree with each other on whwat is archived.

I was looking for example at https://github.com/mautrix/facebook/issues/161 and some people outright leave rooms when they don't want to see them or are done with them for the time being, and have explicitly said they are doing it to have an email-like workflow.

This feature seems to be already available in the proprietary extended Matrix known as Beeper, but I could not find any MSC describing this:

![image](https://github.com/matrix-org/matrix-spec/assets/24363938/da78006f-fbcc-41e6-996f-9bad01dfc606)

This client seems to use a proprietary `com.beeper.inbox.done` room account data.

![image](https://github.com/matrix-org/matrix-spec/assets/24363938/e8a22f65-9c6c-4a64-9f12-655815f13373)

This is how the UI looks like, with only unarchived messages showing up in the "Inbox".

Some issues with this could be that there may be multiple ideas on implementation or how the feature should work. For example, WhatsApp is configurable so you can either:

1) all archived chats are chats I don't really want to see unless I go to the archive
2) I am archiving chats because I don't want to see them for the time being (I have responded, or I don't need to respond, etc) but will need to see it again once new messages show up

I believe 1) should already be met by the "low priority" tag implemented by Element and mentioned in the spec:

```json
{
  "type": "m.tag",
  "content": {
    "tags": {
      "m.lowpriority": {}
    }
  }
}
```

so this issue would be about some sort of `m.archive` tag that disappears when a new message is received.

**Comment by turt2live** [comment]
By default in the specification a room is "archived" upon leaving. It must be *forgotten* to be removed from that archive.

Some servers treat leaving as forgetting, enabling cleanup functions.

**Comment by turt2live** [comment]
(there's also Low Priority tags and Spaces which can help organize/temporarily hide rooms)

**Comment by HammyHavoc** [comment]
> (there's also Low Priority tags and Spaces which can help organize/temporarily hide rooms)

Have you tried the archive function of WhatsApp?

It will hide something until it gets a new message as an option. Low priority and Spaces aren't a good solution. I've consolidated all my chat apps via bridges into Matrix, and navigating my list of conversations is terrible.

**Comment by gabrc52** [comment]
What the spec calls "archived" Element (for example) calls "historical", and it is a different concept from the one suggested in this issue (rooms you've left or been kicked out of).

In the spec, it might need to be called something different from "archived", but it would still make sense for some client implementations to call it "archive".

Putting rooms on a specific space would have the same drawback as "low priority" since they wouldn't be taken out of the space when a new message is received.

**Comment by DarkPhyber-hg** [comment]
Adding support for this.  It makes sense for direct chats, or bridged chats (sms, google chat, fb/ig dm's, etc).  Once a conversation is done, i may not want to "leave" it, but i also don't want to see it, it's distracting.  I try to use a zero inbox philosophy with my chats/messages.  If i need to do something i leave the chat visible, like my wife texted me to pick up milk.  Otherwise, i hide/archive it to have less clutter.

**Comment by HammyHavoc** [comment]
> Adding support for this. It makes sense for direct chats, or bridged chats (sms, google chat, fb/ig dm's, etc). Once a conversation is done, i may not want to "leave" it, but i also don't want to see it, it's distracting. I try to use a zero inbox philosophy with my chats/messages. If i need to do something i leave the chat visible, like my wife texted me to pick up milk. Otherwise, i hide/archive it to have less clutter.

Would also be nice to fav/star/"save" (ala MS Teams) individual messages to be able to come back to them to reply/action/reference. But yes, archiving is a must! My list room list is quite out of hand.

**Comment by gabrc52** [comment]
>  If i need to do something i leave the chat visible

What I usually do is that if I need to do something I mark it as unread, but that is also missing from Matrix... https://github.com/matrix-org/matrix-spec/issues/612
